File: pathfinder-api/src/main/java/de/cubbossa/pathapi/group/PermissionModifier.java
package de.cubbossa.pathapi.group;

import de.cubbossa.pathapi.misc.NamespacedKey;

public interface PermissionModifier extends Modifier {

  NamespacedKey KEY = NamespacedKey.fromString("pathfinder:permission");

  @Override
  default NamespacedKey getKey() {
    return KEY;
  }

  String permission();
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/group/PermissionModifier.kt
package de.cubbossa.pathfinder.group;

import de.cubbossa.pathfinder.misc.NamespacedKey;

public interface PermissionModifier extends Modifier {

  NamespacedKey KEY = NamespacedKey.fromString("pathfinder:permission");

  @Override
  default NamespacedKey getKey() {
    return KEY;
  }

  String permission();
}
